Tooltip.jsx

```diff
@@ -3,178 +3,149 @@
-        // re-render once it's loaded
-        RcTooltip => {
-          // eslint-disable-next-line flexport/no-setState-in-constructor
-          this.setState({RcTooltip});
-        },
-        error => {
-          if (!(error && error.isCanceled)) {
-            throw error;
-          }
-        }
-      );
-
-      if (props.visible === true) {
-        beginLoadingRcTooltip();
-      }
-    }
-    this.state = {RcTooltip};
-  }
-
-  componentWillUnmount() {
-    if (this.waitForRcTooltip !== null) {
-      this.waitForRcTooltip.cancel();
-      this.waitForRcTooltip = null;
-    }
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps.visible !== true && this.props.visible === true) {
-      beginLoadingRcTooltip();
-    }
-  }
-
-  handleMouseEnter = () => {
-    this.defaultVisible = true;
-    beginLoadingRcTooltip();
-  };
-
-  handleMouseLeave = () => {
-    this.defaultVisible = false;
-  };
-
-  render() {
-    const {RcTooltip} = this.state;
-
-    // When RcTooltip isn't loaded, render the child and detect hovering
-    if (RcTooltip === null) {
-      return React.Children.map(this.props.children, child => {
-        if (React.isValidElement(child)) {
-          const {
-            props: {onMouseEnter, onMouseLeave},
-          } = child;
-          // Hello. If you're here because the invariant below triggered, you
-          // can add support for what you are doing by chaining the handlers on
-          // children in the cloneElement call. This use case didn't exist when
-          // I wrote this component so I didn't support it.
-          invariant(
-            onMouseEnter == null && onMouseLeave == null,
-            `Defining onMouseEnter and onMouseLeave on children of Tooltip is
-            currently not implemented! See this error in the source for more
-            info`
-          );
-        }
-        return React.cloneElement(child, {
-          onMouseEnter: this.handleMouseEnter,
-          onMouseLeave: this.handleMouseLeave,
-        });
-      });
-    }
-
-    // If the user is hovering over the child element, and code has just
-    // arrived, the tooltip should be immediately visible
-    const defaultVisible =
-      this.props.defaultVisible !== undefined
-        ? this.props.defaultVisible
-        : this.defaultVisible;
-
-    return <RcTooltip {...this.props} defaultVisible={defaultVisible} />;
+function getPlacementStyles(placement: string) {
+  switch (placement) {
+    case "top":
+      return styles.downArrow;
+    case "bottom":
+      return styles.upArrow;
+    case "left":
+      return styles.rightArrow;
+    case "right":
+      return styles.leftArrow;
+    default:
+      return null;
   }
 }
+
+const arrowVerticalOffset = "-14px";
+
+const styles = StyleSheet.create({
+  trigger: {
+    display: "inline-block",
+  },
+  tooltipContainer: {
+    display: "flex",
+    flexDirection: "column",
+    margin: "0.4rem",
+    transition: "opacity 0.3s",
+  },
```

Comment:
Tooltip should have a z-index of 1500 so it rides over Modals
```suggestion
    zIndex: zIndices.zIndex1500AboveModal.value
  },
```

![image](https://user-images.githubusercontent.com/5897853/67989586-87c3f500-fbf0-11e9-83a2-8f5270fd6475.png)